graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -0,0 +1,320 @@
+            : ImmutableList.of(
+                BuiltCondition.of(LATEST_VERSION_COLUMN_NAME, Predicate.EQ, expectedLatestVersion),
+                BuiltCondition.of(DEPLOYMENT_IN_PROGRESS_COLUMN_NAME, Predicate.NEQ, true));
+    BoundQuery updateDeploymentToInProgress =
+        dataStore
+            .queryBuilder()
+            .update(KEYSPACE_NAME, TABLE_NAME)
+            .value(DEPLOYMENT_IN_PROGRESS_COLUMN_NAME, true)
+            .where(NAMESPACE_COLUMN_NAME, Predicate.EQ, namespace)
+            .ifs(conditions)
+            .build()
+            .bind();
+
+    ResultSet resultSet = dataStore.execute(updateDeploymentToInProgress).get();
+    Row row = resultSet.one();
+    if (!row.getBoolean(APPLIED_COLUMN_NAME)) {
+      boolean hasVersion =
+          row.columns().stream().anyMatch(c -> LATEST_VERSION_COLUMN_NAME.equals(c.name()));
+      if (!hasVersion) {
+        throw new IllegalStateException(
+            "You specified expectedVersion but no previous version was found");
+      }
+      UUID actualLatestVersion = row.getUuid(LATEST_VERSION_COLUMN_NAME);
+      if (Objects.equals(actualLatestVersion, expectedLatestVersion)) {
+        assert row.getBoolean(DEPLOYMENT_IN_PROGRESS_COLUMN_NAME);
+        throw new IllegalStateException(
+            "It looks like someone else is deploying a new schema, please check the latest version and try again. "
+                + "This can also happen if a previous deployment failed unexpectedly, in that case you can use the "
+                + "'force' argument to bypass this check.");
+      }
+      throw new IllegalStateException(
+          String.format(
+              "You specified expectedVersion %s, but there is a more recent version %s",
+              expectedLatestVersion, actualLatestVersion));
+    }
+  }
+
+  public void abortDeployment(String namespace) throws ExecutionException, InterruptedException {
+    BoundQuery updateDeploymentToNotInProgress =
+        dataStore
+            .queryBuilder()
+            .update(KEYSPACE_NAME, TABLE_NAME)
+            .value(DEPLOYMENT_IN_PROGRESS_COLUMN_NAME, false)
+            .where(NAMESPACE_COLUMN_NAME, Predicate.EQ, namespace)
+            .build()
+            .bind();
+    dataStore.execute(updateDeploymentToNotInProgress).get();
+  }
+
+  public void purgeOldSchemaEntries(String namespace) throws Exception {
+    List<SchemaSource> allSchemasForNamespace = getSchemaHistory(namespace);
+
+    int numberOfEntriesToRemove =
+        allSchemasForNamespace.size() - NUMBER_OF_RETAINED_SCHEMA_VERSIONS;
+    if (numberOfEntriesToRemove > 0) {
+      LOGGER.info("Removing {} old schema entries.", numberOfEntriesToRemove);
+
+      // remove N oldest entries
+      SchemaSource mostRecentToRemove =
+          allSchemasForNamespace.get(NUMBER_OF_RETAINED_SCHEMA_VERSIONS);
+
+      BoundQuery deleteSchemaQuery =
+          dataStore
```

Comment:
All of the references to datastore in here are an "internal" datastore aren't they? These queries all need to be in the context of an authenticatedUser so that the correct keyspace is used.

Reply:
Hrmm I need to check. Things should be fine in the fetchers, but there might be some issues in `SchemaFirstCache`.  We're planning to merge the two caches anyway, so it will be a good moment to revisit this.

Reply:
Just to confirm, I'm working on the merge now and this will indeed get fixed in the process.